packages/edit-site/src/components/editor/use-resolve-edited-entity.js

```diff
@@ -187,18 +72,22 @@ export function useResolveEditedEntity() {
 				postId &&
 				authorizedPostTypes.includes( postType )
 			) {
-				return resolveTemplateForPostTypeAndId( postType, postId );
+				return getTemplateId( postType, postId );
 			}
 
 			// If we're rendering the home page, and we have a static home page, resolve its template.
-			if ( homepageId ) {
-				return resolveTemplateForPostTypeAndId( 'page', homepageId );
+			if ( homepagePostType === 'page' ) {
+				return getTemplateId( 'page', homepagePostId );
+			}
+
+			if ( homepagePostType === 'wp_template' ) {
+				return homepagePostId;
 			}
 
-			// If we're not rendering a specific page, use the front page template.
-			return getDefaultTemplateId( { slug: 'front-page' } );
+			// This can only happen when homepage is being loaded.
+			return null;
```

Comment:
This should be an early return. Then all the code that follows has better types: it doesn't need to deal with `undefined` or `null` values.

Reply:
not really, this is only important for the case where we're resolving the home page, for all the other case, there's no need to wait for the home page to actually load or not.

Reply:
Ok I see it now. It can be moved before the `if ( homepagePostType === 'page' )` check, but not earlier.

Another thing I just noticed is that the `resolvedTemplateId = useSelect` callback sometimes returns `undefined` and sometimes `null`. Should be unified.